lib/ansible/plugins/action/ce.py

```diff
@@ -93,10 +93,14 @@ def run(self, tmp=None, task_vars=None):
                 socket_path = self._connection.socket_path
             conn = Connection(socket_path)
             out = conn.get_prompt()
-            while to_text(out, errors='surrogate_then_replace').strip().endswith(']'):
+            prompt = to_text(out, errors='surrogate_then_replace').strip()
+            while prompt.endswith(']'):
                 display.vvvv('wrong context, sending exit to device', self._play_context.remote_addr)
+                if prompt.startswith('[*'):
+                    conn.exec_command('clear configuration candidate')
                 conn.exec_command('return')
```

Comment:
Return command will discard configure directly and return to user-view, which is the base context.
But some new version os will give a prompt to wait a confirm command with a return until it trigger a time out error.
So fix it.